core-blocks/paragraph/index.js

```diff
@@ -233,11 +233,17 @@ class ParagraphBlock extends Component {
 						} }
 						onSplit={ insertBlocksAfter ?
 							( before, after, ...blocks ) => {
-								setAttributes( { content: before } );
-								insertBlocksAfter( [
-									...blocks,
-									createBlock( 'core/paragraph', { content: after } ),
-								] );
+								if ( after ) {
+									blocks.push( createBlock( name, { content: after } ) );
+								}
+
```

Comment:
Unfortunately, we have a problem here on undo insertBlocksAfter and setAttributes/onReplace are separate actions both of them undoable. So undo behaves in a strange way. This was an already existing problem, but seeing the diff reminded me of it.

Reply:
Yeah. This is not new. Would be interesting to just do one `onReplace` call. The problem with that is selection, which should be placed on the second block then.

Reply:
Using onReplace is an awesome idea 👍 We can try to add a property to the replace action that allows us to choose the block that is going to be selected.
I can try to wrangle something around this idea.